neuralprophet/forecaster.py

```diff
@@ -1985,22 +2015,27 @@ def plot_components(
             forecast_in_focus=forecast_in_focus,
         )
 
-        # Check whether the default plotting backend is overwritten
+        # Check whether a local or global plotting backend is set.
         plotting_backend = (
-            plotting_backend
-            if plotting_backend != "default"
-            else (self.plotting_backend if hasattr(self, "plotting_backend") else "matplotlib")
+            auto_set_plotting_backend(plotting_backend)
+            if plotting_backend is not None
+            else (
+                auto_set_plotting_backend(self.plotting_backend)
+                if hasattr(self, "plotting_backend")
+                else auto_set_plotting_backend("plotly-auto")
+            )
```

Comment:
Suggestion: For ease of reading, you could use a more traditional if-else format.
Also, this block seems repeated. maybe part of it can be simplified of part of the auto-setter?